File: Sorting_algorithms/python/bubble_sort.py
def bubble_sort(arr:list):
    arr_len = len(arr)
    for i in range(0, arr_len - 1):
        for j in range(0, arr_len - 1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr

if __name__ == "__main__":
    arr = [11, 30, -1, 2, 7, 12, 13, 20, 8]
    print(f"\nBefore Sort: {arr}")
    print(f"After Sort: {bubble_sort(arr=arr)}\n")

File: Sorting_algorithms/Cpp/src/bubble_sort.cpp
#include <iostream>

using namespace std;

void bubbleSort(int arr[], int arr_lenght)
{
    for (int i = 0; i < arr_lenght - 1; i++)
    {
        for (int j = 0; j < arr_lenght - 1; j++)
        {
            if (arr[j] > arr[j+1])
            {
                swap(arr[j], arr[j+1]);
            }
        }
    }
}


void printArray(int arr[], int arr_lenght)
{
    for (int i = 0; i < arr_lenght; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main()
{
    int arr[] = {11, 30, -1, 2, 7, 12, 13, 20, 8};
    int arr_lenght = sizeof(arr) / sizeof(arr[0]);
    cout << endl;
    cout << "Before sort: ";
    printArray(arr, arr_lenght);
    bubbleSort(arr, arr_lenght);
    cout << "After sort: ";
    printArray(arr, arr_lenght);
    cout << "\n ";
    return 0;
}